docs/howto/openstack/neutron/vpnaas.md

```diff
@@ -0,0 +1,586 @@
+    | ID             | bb1a307d-6f8f-4a0a-83db-7c705403485d |
+    | Name           | vpn-service-kna1                     |
+    | Project        | 94109c764a754e24ac0f6b01aef82359     |
+    | Router         | 5ac45739-a379-4936-8b1b-67d10e017f4d |
+    | State          | True                                 |
+    | Status         | PENDING_CREATE                       |
+    | Subnet         | None                                 |
+    | external_v4_ip | 203.0.113.101                        |
+    | external_v6_ip | 2a00:16d8:0:3b:f816:3eff:fe0e:2074   |
+    | project_id     | 94109c764a754e24ac0f6b01aef82359     |
+    +----------------+--------------------------------------+
+    ```
+    
+    For convenience, set the value of parameter `external_v4_ip ` to a 
+    shell variable:
+        
+    ```bash
+    EXT_IP_KNA1="203.0.113.101"
+    ```
+        
+    Create a local end-point group:
+     
+    ```bash
+    openstack vpn endpoint group create \
+        --type subnet --value subnet-kna1 local-epg-kna1
+    ```
+    
+    ```plain
+    +-------------+------------------------------------------+
+    | Field       | Value                                    |
+    +-------------+------------------------------------------+
+    | Description |                                          |
+    | Endpoints   | ['421d8fd2-dd7f-4f7c-9a51-42ef4a866dd9'] |
+    | ID          | c1937c3d-77e7-4f2c-842d-70b5e10df9a8     |
+    | Name        | local-epg-kna1                           |
+    | Project     | 94109c764a754e24ac0f6b01aef82359         |
+    | Type        | subnet                                   |
+    | project_id  | 94109c764a754e24ac0f6b01aef82359         |
+    +-------------+------------------------------------------+
+    ```
+    
+    Create a peer (remote) end-point group:
+    
+    ```bash
+    openstack vpn endpoint group create \
+        --type cidr --value $SUBNET_FRA1 peer-epg-kna1
+    ```
+    
+    ```plain
+    +-------------+--------------------------------------+
+    | Field       | Value                                |
+    +-------------+--------------------------------------+
+    | Description |                                      |
+    | Endpoints   | ['10.15.25.0/24']                    |
+    | ID          | 2e627315-02f0-4d68-8683-14230b166060 |
+    | Name        | peer-epg-kna1                        |
+    | Project     | 94109c764a754e24ac0f6b01aef82359     |
+    | Type        | cidr                                 |
+    | project_id  | 94109c764a754e24ac0f6b01aef82359     |
+    +-------------+--------------------------------------+
+    ```
+    
+    ### Instantiate a pre-shared key
+       
+    Before establishing a site-to-site IPSec VPN connection, you must 
+    have a randomly generated pre-shared key. You may use `openssl` for 
+    generating a random string and immediately set it to a shell variable:
+        
+    ```bash
+    PRE_SHARED_KEY=$(openssl rand -hex 24)
```

Comment:
Does this _have_ to be a 24-byte hex key? If it does not, then please explain that this is an example and that you could also use, say, `pwgen 64 1` for the same purpose.

Reply:
No, no, it does not have to be a 24-byte hex key -- that's just an example I came up with, God-knows-why. I will clarify that and also mention (the fine) `pwgen`, which I use and love anyways.